src/search/lp/lp_solver.cc

```diff
@@ -70,7 +70,11 @@ void LPConstraint::insert(int index, double coefficient) {
     coefficients.push_back(coefficient);
 }
 
-ostream &LPConstraint::dump(ostream &stream, double infinity, const LinearProgram *program) {
+ostream &LPConstraint::dump(ostream &stream, const LinearProgram *program) {
+    double infinity = numeric_limits<double>::infinity();
+    if (program) {
+        infinity = program->get_infinity();
+    }
```

Comment:
I don't fully understand what this does, but if `infinity` can potentially have different values, is it guaranteed that `-infinity` in the following will do the right thing? Just as an example, in an integer world (which I know we don't have, but just as an example since clearly it looks like there are different possible representations of infinity involved), typical representations of infinity don't necessarily allow negating by just using `-`.

Reply:
Yes, according the OSI interface, you are supposed to use `-solver.get_infinity()` for negative infinity. The fallback to `numeric_limits<double>::infinity()` is just used for dumping the constraints as debug output without specifying a solver. The use case for this is debugging the creation of a model where you might be in a nested function call that does not have direct access to the solver, or does not yet have the `LinearProgram` class constructed. In that case, we cannot print the constraint as nicely (no variable names, no infinity value) but we can still print some representation of it.